frontend/javascripts/admin/task/task_search_form.js

```diff
@@ -95,7 +96,11 @@ class TaskSearchForm extends React.Component<Props, State> {
     this.setState({ users, projects, taskTypes });
   }
 
-  handleFormSubmit = (isRandom: boolean, event: ?SyntheticInputEvent<*>) => {
+  handleFormSubmitWithSubsequentCall = (
+    isRandom: boolean,
+    subsequentMethod: QueryObject => Promise<void>,
```

Comment:
`subsequentMethod` is quite an unusual name, in my opinion. I'd go with `onFinish`, `onFinishCallback` or `onFinishFunction` instead. Also, the method name can remain at `handleFormSubmit`, since the method name doesn't need to explain its parameters.

Reply:
As you noticed correctly, I had problems with finding a good naming here. :confused: 
The reason why I renamed the `handleFormSubmit` method is, that this name was already used in line 142.

But in my opinion, the naming should be fine now in the newest changes. :+1: